packages/quantic/force-app/main/default/lwc/quanticSort/quanticSort.js

```diff
@@ -150,4 +207,31 @@ export default class QuanticSort extends LightningElement {
   get value() {
     return this.state?.sortCriteria;
   }
+
+  /**
+   * Returns an array of custom sort options passed via slots.
+   * @returns {SortOption[]} The specified custom sort options.
+   */
+  get customSortOptions() {
+    /** @type {SortOption[]} */
+    // @ts-ignore
+    const elements = Array.from(this.querySelectorAll('c-quantic-sort-option'));
+    if (elements.length === 0) {
+      return [];
+    }
+
+    return elements.map((option) => {
+      return {
+        label: option.label,
+        value: option.value,
+        criterion: {
+          by: option.criterion.by,
+          // @ts-ignore
+          order: option.criterion.order,
+          // @ts-ignore
+          field: option.criterion.field,
+        },
+      };
+    });
```

Comment:
```suggestion
    return elements.map((option) => ({
      label,
      value,
      criterion
    }));
```
That should work AFAIK?

Reply:
I can just do this also, simple and clean:

<img width="317" alt="image" src="https://github.com/coveo/ui-kit/assets/73316533/4c8a8596-e7c5-4a03-93d3-7e4cad340866">